compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -2714,6 +2718,27 @@ impl<'a> Parser<'a> {
         if let ExprKind::Let(_, _, _, Recovered::No) = cond.kind {
             // Remove the last feature gating of a `let` expression since it's stable.
             self.psess.gated_spans.ungate_last(sym::let_chains, cond.span);
+        } else {
+            fn ungate_let_exprs(this: &mut Parser<'_>, expr: &Expr) {
+                if !expr.span.at_least_rust_2024() {
+                    return;
+                }
+                match &expr.kind {
+                    ExprKind::Binary(BinOp { node: BinOpKind::And, .. }, lhs, rhs) => {
+                        ungate_let_exprs(this, rhs);
+                        ungate_let_exprs(this, lhs);
+                    }
+                    ExprKind::Let(..) => {
+                        this.psess.gated_spans.ungate_last(sym::let_chains, expr.span)
+                    }
+                    _ => (),
+                }
+            }
+            if edition.at_least_rust_2024() {
+                // Scoping code checks the top level edition of the `if`: let's match it here.
+                // Also check all editions in between, just to make sure.
```

Comment:
```suggestion
                // Scoping code checks the top level edition of the `if`; let's
                // match it here.  Also check all editions in between, just to
                // make sure.
```

(Replacing colon with semicolon as appropriate here.)

Reply:
Put a ; there.